dash/include/dash/Array.h

```diff
@@ -1014,7 +1066,10 @@ class Array
   void barrier() const
   {
     DASH_LOG_TRACE_VAR("Array.barrier()", m_team);
-    if (nullptr != m_team && dash::Team::Null() != *m_team) {
+    if (nullptr != m_globmem) {
+      m_globmem->flush_all();
+    }
+    if (nullptr != m_team && *m_team != dash::Team::Null()) {
```

Comment:
Currently yes, but it's a pointer, so I check it. Members can be specified by *lots* of operations and checking for `nullptr` is more robust than validating every possible member assignment.
Also, it's in `Array::barrier()` so it's not like the test will have a say in performance.